pages/builds/writing_build_scripts.md.erb

```diff
@@ -1,90 +1,140 @@
 # Writing Build Scripts
 
-Steps in your build pipelines are most often just shell scripts that are checked in alongside your code. The [Buildkite Agent](/docs/agent) will run your scripts, capture and report the log output, and use the exit status to mark the build as passed or failed.
+One of the most common actions that Buildkite steps perform is running shell scripts. These scripts are checked in alongside your code and `pipeline.yml` file. The [Buildkite Agent](/docs/agent) will run your scripts, capture and report the log output, and use the exit status to mark each job, as well as the overall build, as passed or failed.
 
 <%= toc %>
 
 ## Configuring Bash
 
-There are four Bash options which are often used in build scripts:
+The shell that runs your scripts in Buildkite is a clean Bash prompt with no settings. If you rely on anything from your `~/.bash_profile` or `~/.bashrc` files when you run scripts locally, you'll need to explicitly add the relevant items to your build scripts.
+
+When writing Bash shell scripts there are a number of options you can set to help prevent unexpected errors:
 
 <table>
   <tbody>
   <tr>
-    <th><code>-e</code></th>
-    <td>Exit immediately if a command exits with a non-zero status</td>
+    <th><code>e</code></th>
+    <td>Exit script immediately if any command returns a non-zero exit status.</td>
   </tr>
   <tr>
-    <th><code>-u</code></th>
-    <td>Exit immediately if an undefined variable is used (e.g. <code>echo "$UNDEFINED_ENV_VAR"</code>)</td>
+    <th><code>u</code></th>
+    <td>Exit script immediately if an undefined variable is used (e.g. <code>echo "$UNDEFINED_ENV_VAR"</code>).</td>
   </tr>
   <tr>
-    <th><code>-o pipefail</code></th>
-    <td>Ensures pipelines (e.g. <code>cmd | othercmd</code>) return a non-zero status if a command fails</td>
+    <th><code>o pipefail</code></th>
+    <td>Ensure Bash pipelines (e.g. <code>cmd | othercmd</code>) return a non-zero status if any of the commands fail, rather than returning the exit status of the last command in the pipeline.</td>
   </tr>
   <tr>
-    <th><code>-x</code></th>
-    <td>Print each expanded command before executing (useful for debugging)</td>
+    <th><code>x</code></th>
+    <td>Expand and print each command before executing. See <a href="/docs/builds/writing-build-scripts#debugging-your-environment">Debugging your environment</a> for more information. </td>
   </tr>
   </tbody>
 </table>
 
-Most build scripts should set the `-e` and `-o pipefail` options to ensure if any command fails your entire build script fails. For example:
+Bash’s built-in `set` command with a `-` argument is used to enable options (the `+` argument is used to disable options).
+
+For example, the following script enables the most commonly used options for build scripts:
 
 ```bash
 #!/bin/bash
 
-set -eo pipefail
+set -euo pipefail
```

Comment:
@lox @sj26 @keithpitt @ticky is this too controversial a suggestion? I know the `set -u` option has caused us grief in the past because quite a few other build tools (rvm comes to mind) blow up with this option enabled. But it's super handy, and I use this as my default all the time.